core/src/main/java/org/elasticsearch/index/query/QueryParseContext.java

```diff
@@ -156,6 +157,11 @@ public SimilarityService similarityService() {
         return indexQueryParser.similarityService;
     }
 
+    @Nullable
+    public ClusterService clusterService() {
```

Comment:
I would leave the ClusterService in IndexQueryParseService so the dependency is more contained there